examples/finding_best_open_source_embedding_model/best_embedding_model_rag_app.ipynb

```diff
@@ -0,0 +1,559 @@
+    "        print(cur.fetchone())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generating Embeddings\n",
+    "\n",
+    "pgai makes generating embeddings for various models incredibly straightforward with its [pgai Vectorizer](https://github.com/timescale/pgai/blob/main/docs/vectorizer.md). With just a single SQL command, you can effortlessly configure a vectorizer to automatically create and update embeddings from any chosen embedding model.\n",
+    "\n",
+    "We create vectorizers for each embedding model. Each vectorizer will:\n",
+    "\n",
+    "- Process text using the same chunking strategy (512 characters with 50 character overlap)\n",
+    "- Use consistent text formatting across all models\n",
+    "- Generate embeddings with model-specific dimensions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_vectorizer(embedding_model, embeddings_dimensions):\n",
+    "    embeddings_view_name = f\"{'essays'}{'_'}{embedding_model.replace('-','_')}{'_'}{'embeddings'}\"\n",
+    "\n",
+    "    with connect_db() as conn:\n",
+    "        with conn.cursor() as cur:\n",
+    "            cur.execute(\"\"\"\n",
+    "                SELECT ai.create_vectorizer(\n",
+    "                'essays'::regclass,\n",
+    "                destination => %s,\n",
+    "                embedding => ai.embedding_ollama(%s, %s),\n",
+    "                chunking => ai.chunking_recursive_character_text_splitter('text', 512, 50),\n",
+    "                formatting => ai.formatting_python_template('title: $title date: $date $chunk')\n",
+    "            );\n",
+    "            \"\"\", (embeddings_view_name, embedding_model, embeddings_dimensions, )\n",
+    "            )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "EMBEDDING_MODELS = [\n",
+    "    {'name':'mxbai-embed-large', 'dimensions': 1024},\n",
+    "    {'name':'nomic-embed-text','dimensions': 768},\n",
+    "    {'name':'bge-m3','dimensions': 1024},\n",
+    "] \n",
+    "\n",
+    "for model in EMBEDDING_MODELS:\n",
+    "    create_vectorizer(model['name'], model['dimensions'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The vectorizers may take some time to complete the embedding generation. Use the following command to monitor their progress:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with connect_db() as conn:\n",
+    "    with conn.cursor() as cur:\n",
+    "        cur.execute(\"SELECT * FROM ai.vectorizer_status;\")\n",
+    "\n",
+    "        for row in cur.fetchall():\n",
+    "            print(f\"Vectorizer ID: {row[0]}, Embedding Table: {row[2]}, Pending Items: {row[4]}\")\n",
```

Comment:
this returns 215 pending for a long time for me and looks like no progress, might make sense to make the user aware that this will take hours on a cpu only machine

Reply:
That's a good suggestion. I will update the text before that to reflect it.